monitoring/src/listener/records.rs

```diff
@@ -0,0 +1,149 @@
+use std::sync::Arc;
+use serde::{Serialize, Deserialize};
+use storage::{
+    persistent::DatabaseWithSchema,
+    StorageError,
+};
+use storage::persistent::{Schema, Codec, SchemaError};
+use byteorder::{LittleEndian, WriteBytesExt, ReadBytesExt};
+
+// --- Storing result in Rocks DB --- //
+pub type RecordMetaStorageDatabase = dyn DatabaseWithSchema<RecordMetaStorage> + Sync + Send;
+pub type RecordStorageDatabase = dyn DatabaseWithSchema<RecordStorage> + Sync + Send;
+
+#[derive(Clone)]
+pub struct RecordMetaStorage {
+    db: Arc<RecordMetaStorageDatabase>,
+}
+
+impl RecordMetaStorage {
+    pub fn new(db: Arc<RecordMetaStorageDatabase>) -> Self {
+        Self { db }
+    }
+
+    #[inline]
+    pub fn put_record_meta(&mut self, ts: f32, record_meta: &RecordMeta) -> Result<(), StorageError> {
+        self.db.put(&RocksStamp(ts), record_meta)
+            .map_err(StorageError::from)
+    }
+}
+
+impl Schema for RecordMetaStorage {
+    const COLUMN_FAMILY_NAME: &'static str = "record_meta_storage";
+    type Key = RocksStamp;
+    type Value = RecordMeta;
+}
+
+#[derive(Clone)]
+pub struct RecordStorage {
+    db: Arc<RecordStorageDatabase>,
+}
+
+impl RecordStorage {
+    pub fn new(db: Arc<RecordStorageDatabase>) -> Self {
+        Self { db }
+    }
+
+    pub fn put_record(&mut self, ts: f32, record: &Vec<u8>) -> Result<(), StorageError> {
+        self.db.put(&RocksStamp(ts), record)
+            .map_err(StorageError::from)
+    }
+}
+
+impl Schema for RecordStorage {
+    const COLUMN_FAMILY_NAME: &'static str = "record_storage";
+    type Key = RocksStamp;
+    type Value = Vec<u8>;
+}
+
+// --- Record implementation --- //
+
+/// Simple timestamp, denoting passed seconds from start of the node
+/// implementing the codec trait for simple database serialization.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct RocksStamp(f32);
```

Comment:
What is the reason behind using `f32` here instead of using `u32` and saying that it represents milliseconds from the start of a node? It will help you to avoid floating point precision problems splus you will be able to use https://doc.rust-lang.org/std/primitive.i32.html#method.from_le_bytes instead of `byteorder` crate functions.

Reply:
It is a mostly space saving solution. Millisecond counting requires 128 bit integral type. 32 bit floating point number is precise enough for this use-case, and Rust have a feature, in standard library, to serialize it with the `f32::to_le_bytes` and `f32::from_le_bytes`. Which is, in my opinion good  compromise.

Reply:
Will be replaced by a couter value.

Reply:
Done in [7631196](https://github.com/simplestaking/tezos-rs/pull/55/commits/7631196433a997eea4bf201dd9dfd47756a54c07)